tests/bluemira/geometry/test_varied_offset.py

```diff
@@ -0,0 +1,142 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2022 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh, J. Morris,
+#                    D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+from typing import Callable
+
+import numpy as np
+import pytest
+from scipy.interpolate import interp1d
+
+from bluemira.geometry.parameterisations import PictureFrame
+from bluemira.geometry.tools import find_clockwise_angle_2d, make_circle
+from bluemira.geometry.varied_offset import varied_offset
+from bluemira.geometry.wire import BluemiraWire
+
+
+class TestVariedOffsetFunction:
+
+    fixtures = [
+        {
+            "wire": PictureFrame(
+                {"ro": {"value": 6}, "ri": {"value": 3}}
+            ).create_shape(),
+            "min_offset": 1,
+            "max_offset": 4,
+            "min_offset_angle": 45,  # degrees
+            "max_offset_angle": 160,  # degrees
+        },
+        {
+            "wire": make_circle(axis=(0, 1, 0)),
+            "min_offset": 1,
+            "max_offset": 3,
+            "min_offset_angle": 90,  # degrees
+            "max_offset_angle": 140,  # degrees
+        },
+    ]
```

Comment:
Probably need some more challenging shapes here; PrincetonD, PolySpline spring to mind (almost the more likely use-case parameterisations).